### ISSUE TITLE
MongoDb traces no longer showing after mongodb driver upgrade

### ISSUE THREAD

**Issue by sattarab** [DataDog/dd-trace-js#2006]
We recently updated our mongodb driver from 3.6.4 to 4.3.1. Now the MongoDb traces are no longer being shown in the Datadog dashboard

**Expected behaviour**
Updating the MongoDb library to 4.3.1 the traces should still work

**Actual behaviour**
Updating the MongoDb library to 4.3.1 the traces have stopped working for MongoDb

**Steps to reproduce**
Set the mongodb library version to 4.3.1 in package.json
```
"mongodb": "4.3.1",
```

**Environment**

* **Operation system:** Amazon Linux
* **Node.js version:** 14.7.3
* **Tracer version:** 2.5
* **Agent version:** 6
* **Relevant library versions:** "mongodb": "4.3.1"

**Comment by rochdev** [comment]
Looking at the [nightly test](https://app.circleci.com/pipelines/github/DataDog/dd-trace-js/6919/workflows/9f1186dd-4c6e-489f-9725-fac6e5080b3b/jobs/864725) it looks like our tests are passing for `4.5.0`. Can you share a reproduction snippet?

**Comment by sattarab** [comment]
@rochdev we are using the following way in our code in coffeescript
```
tracer = require( 'dd-trace' ).init {
    analytics: true
    logInjection: true
    runtimeMetrics: true
    service: "web"
    trackAsyncScope: false
    tags:
      app: "web"
      env: "staging"
  }

  tracer.use 'http', {
    blacklist: ignored_routes
  }

  tracer.use 'express', {
    blacklist: ignored_routes
  }
```
Note that some of the options are outdated here as we were using version 0.31 of the library prior when mongodb traces was working for us

**Comment by rochdev** [comment]
Can you share runnable code that reproduces the issue? On my end everything works as it should.

**Comment by sattarab** [comment]
@rochdev You can use the following code to reproduce
**app.js**
```
const { tracer } = require('dd-trace');

tracer.init({
  env: 'dev-sattarab',
  service: 'mongodb-dev',
  version: 'dev-sattarab'
});

const { MongoClient } = require('mongodb');

const url = 'mongodb://localhost:27017';
const client = new MongoClient(url);
const dbName = 'myProject';

async function main() {
  // Use connect method to connect to the server
  await client.connect();
  console.log('Connected successfully to server');
  const db = client.db(dbName);
  const collection = db.collection('documents');
  await collection.insertOne( { name: "test" } )
  const count = await collection.countDocuments()
  const documents = await collection.find().toArray()
  console.log( "documents", documents )
  console.log('count', count)

  return 'done.';
}

main()
  .then(console.log)
  .catch(console.error)
  .finally(() => client.close());
```

**package.json**
```
{
  "name": "mongo-tracer",
  "dependencies": {
    "dd-trace": "2.5.0",
    "mongodb": "4.5.0"
  }
}
```

Can run like `node app.js`

**Note** I tried it with Datadog agent 7 as well

Screenshot 

<img width="1721" alt="Screen Shot 2022-04-28 at 12 14 47 PM" src="https://user-images.githubusercontent.com/5460756/165797761-651d2a8b-ef01-4598-938e-628face9ada5.png">

**Comment by sattarab** [comment]
@rochdev any update on this?

**Comment by rochdev** [comment]
I've just tried the above code locally and I'm getting traces. Could it be related to the CoffeeScript transpilation? Did you try the above pure JavaScript version as well?

**Comment by sattarab** [comment]
@rochdev I did try the above javascript code that I shared with you. The above screenshot is what I got. I am not sure what exactly is wrong? Could you share what you did and the screenshot?

**Comment by rochdev** [comment]
The steps I took:

* Create `app.js` and `package.json` with the above code in a new folder.
* Run `yarn install` in the folder.
* Run `DD_TRACE_DEBUG=true node app`

I get this output (there are more spans but this would be the relevant one):

```
[{"trace_id":"0d0383f7a9007798","span_id":"0d0383f7a9007798","parent_id":"0000000000000000","name":"mongodb.query","resource":"find myProject.documents {}","error":0,"meta":{"service":"mongodb-dev","env":"dev-sattarab","version":"dev-sattarab","runtime-id":"4b5db7f7-45fa-4e3d-b59d-d7933dd45bc4","span.kind":"client","db.name":"myProject.documents","mongodb.query":"{}","out.host":"127.0.0.1","out.port":"27017"},"metrics":{"_dd.agent_psr":1,"_dd.measured":1,"_sampling_priority_v1":1},"start":1651696583032001500,"duration":2357666,"service":"mongodb-dev-mongodb","type":"mongodb"}]
```

This is what I see in the UI:

<img width="822" alt="image" src="https://user-images.githubusercontent.com/1596303/166822972-99012a76-739a-4917-978a-8bddc8e37d4c.png">

Not the prettiest flame graph as queries would usually happen during a request, but I do get the trace. I tried with Node 14 as well to be sure, and it still works. Not sure what else could the issue. Can you reproduce in Docker maybe to remove anything specific to the environment?

**Comment by apank** [comment]
@sattarab - I'm having the same issue with mongoose after updating dd-trace.

For me if I use dd-trace 2.2.0 all my mongo traces are logged; but if I upgrade any higher the mongo traces stop being ingested.

**Comment by guspio** [comment]
Hi!

With version 2.7.0, I have the APM traces of MongoDB again!

Thanks

**Comment by sattarab** [comment]
@guspio doesn't work with `2.7.0` for us

**Comment by apank** [comment]
> @rochdev It works when I downgrade it to 2.2 so I am going to do that for now. @apank thankyou so much for the suggestion

np, it's very frustrating - with every release they break more things - there's other integrations that don't get pulled in anymore in the newer versions such as bunyan

**Comment by rochdev** [comment]
@apank Can you provide a reproduction? So far every reproduction that was provided was working as expected.

**Comment by nicolas-vivot** [comment]
Also it looks like the environment variable DD_TRACE_DISABLED_PLUGINS to specifically disabled some plugins is completely ignored by the newest versions..

**Comment by ondrejnebesky** [comment]
Bumped into this issue with dd-trace@2.9.1 and mongodb@3.6.20 on NodeJS 16. Downgrade to 2.3.0 resolves the issue, but I would rather use the current version. I'll post simple case here when ready.

**Comment by sattarab** [comment]
@apank have you tried it with mongodb 4.7.0?

**Comment by apank** [comment]
I'm using mongoose v5.13.14 which use mongodb v3.7.3

I tested the latest version of dd-trace  (2.10.0) - it didn't capture any of the mongo traces at all :(

**Comment by avivshafir** [comment]
any update on this guys?

**Comment by avivshafir** [comment]
wasn't working for me either, locked versions to 
"mongodb": "4.4.1", and "dd-trace": "2.5.0" then it started working.

**Comment by apank** [comment]
I use mongoose

**Comment by apank** [comment]
@avivshafir @sattarab @guspio @ondrejnebesky @nicolas-vivot 

Great news!

A few weeks ago I was able to try @rochdev's suggestion above and everything has been working as expected for me in v3.9.3 😎 😎 😎 

I did the following - 
In my app I created a file named `datadog.js` in my app's `src/config` directory. 
```
const tracer = require('dd-trace');
const { ENV } = process.env;

tracer.init({
  service: 'omega',
  version: 'alpha.beta.meta',
  logInjection: true,
  profiling: true,
  runtimeMetrics: true,
  startupLogs: true,
  appsec: true,
  env: ENV,
});
tracer.use('mongodb-core', {
  service: 'delta-db',
  measured: true,
});
// other plugin configs etc....
```

Then in `package.json` I updated the `scripts` object `start` property to first require the `./src/config/datadog.js`
The `-r` [cli arg](https://nodejs.org/api/cli.html#-r---require-module) in node ensures that the module is preloaded at startup.

```
"scripts": {
    "start": "node -r ./src/config/datadog.js src/index.js",
}
```


In my prior configuration; `dd-tracer` was  being initialized in `src/index.js` like so below. When I would update `dd-tracer` to any version higher than [v2.2.0](https://github.com/DataDog/dd-trace-js/tree/v2.2.0) - mongo and other traces would not come through. 

```
const express = require('express');
const tracer = require('dd-trace');
const routes = require('./routes/');
const { ENV } = process.env;

if (config.env === 'production') {
  tracer.init({
    service: 'omega',
    version: 'alpha.beta.meta',
    logInjection: true,
    profiling: true,
    runtimeMetrics: true,
    startupLogs: true,
    appsec: true,
    env: ENV,
  });
}

const app = express();
```

When I initially read the [instrumentation docs](https://docs.datadoghq.com/tracing/trace_collection/dd_libraries/nodejs/?tab=containers#instrument-your-application) (in hindsight, not to closely obviously 😛) - I believed since I was importing the library and initializing before initiating the express server itself that the above was valid. Also - this setup worked as expected for [v2.2.0](https://github.com/DataDog/dd-trace-js/tree/v2.2.0) and below.


@rochdev - thank you very much for all your assistance an patience on this matter.
